Refactor landing page and add pages for each usecase.

romanowski opened the pull request.

romanowski committed:
Refactor landing page and add pages for each usecase.

lwronski commented on website/src/components/basicInstall.js:
```diff
@@ -0,0 +1,30 @@
+        {label: 'macOS/Linux', value: 'mac/linux'},
+      ]}>
+        <TabItem value="win">
+          <a className="no_monospace" href="https://github.com/VirtuslabRnD/scala-cli/releases/download/nightly/scala-cli-x86_64-pc-win32.msi">
```
```suggestion
          <a className="no_monospace" href="https://github.com/Virtuslab/scala-cli/releases/download/nightly/scala-cli-x86_64-pc-win32.msi">
```

lwronski commented on website/src/components/features.js:
```diff
@@ -0,0 +1,62 @@
+      <p>
+        scala-cli is built on top of coursier
+        <br/>
+        This allow us to manage Scala versions, dependencies and JVMs so you can test your code in different environments by changing single option.
```
I have doubt it should be here 'single option', because it requires changing a few options, not just one

alexarchambault replied:
One option is often enough, isn't it? Like `--jvm 11`, `--scala-js`, or `--scala 2.13`.

lwronski commented on website/src/components/features.js:
```diff
@@ -0,0 +1,62 @@
+        If you want to use older <b>version of Scala</b> or
+        run your code in <b>JS</b> or <b>Native</b> environments we've got you covered.
+        <br/>
+        <i>some additional <a href="TODO?">setup</a> may be required for JS and Native</i>
```
For scalaJs, the Node is required - https://github.com/VirtuslabRnD/scala-cli-features/issues/6

romanowski replied:
Could you add documentation what and how should be installed to js section in the documentation and link it here? Do we need any specific setup for native?

alexarchambault commented:
The monospaced font _everywhere_ makes it quite hard to read the descriptions, to me. It's fine for titles / headers, but I'm not sure for whole paragraphs.

alexarchambault commented:
Maybe we should agree on a precise convention to write Scala CLI. To me, when referring to the command itself, I write `scala-cli` (usually as code), and when referring to the project, I write Scala CLI (in two words, with Scala capitalized, and CLI in upper case).

alexarchambault commented:
Total NIT: I'm wondering how doable it would be to have two versions of the terminal GIFs, one for dark mode, one for light mode.

alexarchambault commented on README.md:
```diff
@@ -124,7 +124,9 @@ that embeds JARs of the scala-cli modules, and downloads their dependencies
-### Website commands
+### Website 
+
+Website uses [Docusaurus](https://v1.docusaurus.io/en/) and under the hood it uses [infima](https://infima.dev/docs/layout/spacing) framework to style the website. 
```
?

```suggestion
The Scala CLI website is built with [Docusaurus](https://v1.docusaurus.io/en/) and uses [Infima](https://infima.dev/docs/layout/spacing) for styling. 
```

alexarchambault commented on website/docusaurus.config.js:
```diff
@@ -16,9 +16,31 @@ module.exports = {
+            },
+            {
+              to: '/scripting',
+              label: 'Scriptting',
```
```suggestion
              label: 'Scripting',
```

alexarchambault commented on website/src/components/basicInstall.js:
```diff
@@ -0,0 +1,30 @@
+          <p>Run the following one-line command in your terminal 
+            <br/>
+            <code> bash
+              curl -sSf https://virtuslab.github.io/scala-cli-packages/scala-setup.sh | sh
```
I would add an `L` here, just in case (that's to have curl follow redirections):
```suggestion
              curl -sSLf https://virtuslab.github.io/scala-cli-packages/scala-setup.sh | sh
```

alexarchambault commented on website/src/components/features.js:
```diff
@@ -0,0 +1,62 @@
+      </p>
+
+      <p>
+        scala-cli install all its depenendencies
```
?
```suggestion
        Scala CLI installs all its dependencies
```

"ships" sounds better too, although, strictly speaking, Scala CLI installs / downloads them on-the-fly rather than being shipped with them…
```suggestion
        Scala CLI ships with all its dependencies
```

alexarchambault commented on website/src/pages/index.js:
```diff
@@ -1,161 +1,151 @@
+
+              <p>Scala-cli does not require any configuration file and all in-file configuration can be overridden by command-line.</p>
+
+              <p>No additional installation or setup of enviroment (like specific CWD) required.</p>
```
What do you mean by "specific CWD"?

romanowski replied:
specific current working directory like whit sbt

alexarchambault commented:
@lwronski FYI, [the CI error](https://github.com/VirtusLab/scala-cli/pull/151/checks?check_run_id=3705751412) seems legit:
```text
[39/39] packager.run 
Exception in thread "main" java.lang.ArrayIndexOutOfBoundsException: 28
	at net.sf.image4j.codec.ico.ICOEncoder.writeAndBitmap(ICOEncoder.java:352)
	at net.sf.image4j.codec.ico.ICOEncoder.write(ICOEncoder.java:219)
	at net.sf.image4j.codec.ico.ICOEncoder.write(ICOEncoder.java:120)
	at net.sf.image4j.codec.ico.ICOEncoder.write(ICOEncoder.java:106)
	at net.sf.image4j.codec.ico.ICOEncoder.write(ICOEncoder.java:41)
	at packager.windows.DefaultImageResizer$.generateIcon(DefaultImageResizer.scala:18)
	at packager.windows.WindowsPackage.$anonfun$build$2(WindowsPackage.scala:17)
	at scala.Option.map(Option.scala:242)
	at packager.windows.WindowsPackage.$anonfun$build$1(WindowsPackage.scala:17)
	at scala.Option.flatMap(Option.scala:283)
	at packager.windows.WindowsPackage.build(WindowsPackage.scala:16)
	at packager.cli.commands.Build$.run(Build.scala:52)
	at packager.cli.commands.Build$.run(Build.scala:15)
	at caseapp.core.app.CaseApp.main(CaseApp.scala:135)
	at caseapp.core.app.CommandsEntryPoint.main(CommandsEntryPoint.scala:112)
	at packager.cli.PackagerCli.main(PackagerCli.scala)
```

(originates from the new logo files?)

MarcinAman commented:
I think i am lacking some form of definition what is a `simple project`? Without an oral explanation from Krzysiek it is super hard to deduce what we call `simple`. Does simple mean only 1 file? Or maybe 1 package / module? Is using scala-js / native also considered simple? 

in fact it would be super nice to have this definition somewhere on the index page. Maybe an asterisk with anchor link would do? Or even a paragraph just for this

romanowski committed:
Apply suggestions from code review

Co-authored-by: Alexandre Archambault <alexarchambault@users.noreply.github.com>
Co-authored-by: Łukasz Wroński <46607934+lwronski@users.noreply.github.com>

romanowski commented:
@alexarchambault I unified how we use scala-cli and Scala CLI and changed the font in the body of the paragraphs.

As for the terminal, we probably need to use something a bit better than manually recording gifs (like asci-cinema) and then we can play with backgrounds etc. We need to update the content of the gifs anyway.

romanowski committed:
Review feedback:

Simple projects -> single-module projects
Unifing naming od Scala CLI
Add border to gifs and images
detect windows automatically.

romanowski committed:
Fix website building

romanowski committed:
Update logo and change site url
